### PR TITLE
Add gometalinter static check

### PR DESCRIPTION
**Pull request by Madhu-1**

### CODE REVIEW

**Comment by Madhu-1** [comment]
@rootfs yeah added DNM (do not merge to commit) , working on fixing the issues)

**Comment by rootfs** [comment]
that's a huge improve, thanks!

**Comment by Madhu-1** [comment]
@rootfs PTAL

File: .travis.yml [hunk 1]
```diff
@@ -7,15 +7,18 @@ branches:
...
 
-go: 1.10.x
+go: 1.11.x
+
+install:
+  - curl -L https://git.io/vp6lP | sh
 
 before_script:
   - GO_FILES=$(find . -iname '*.go' -type f | grep -v /vendor/)
   - go get -u golang.org/x/lint/golint #go get github.com/golang/lint/golint
```
Comment (rootfs) [inline .travis.yml]: do we still need golint?

Comment (Madhu-1) [reply .travis.yml]: sorry forgot to remove it

File: pkg/cephfs/nodeserver.go [hunk 2]
```diff
@@ -125,30 +128,42 @@ func (ns *NodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
...
+	if err = ns.mount(volOptions, req); err != nil {
+		return nil, err
+	}
+
+	glog.Infof("cephfs: successfully mounted volume %s to %s", volID, stagingTargetPath)
+
+	return &csi.NodeStageVolumeResponse{}, nil
+}
+
+func (*NodeServer) mount(volOptions *volumeOptions, req *csi.NodeStageVolumeRequest) error {
```
Comment (rootfs) [inline pkg/cephfs/nodeserver.go]: is the refactoring required?

File: pkg/cephfs/nodeserver.go [hunk 3]
```diff
@@ -241,7 +265,3 @@ func (ns *NodeServer) NodeGetCapabilities(ctx context.Context, req *csi.NodeGetC
 		},
 	}, nil
 }
-
-func (ns *NodeServer) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
```
Comment (rootfs) [inline pkg/cephfs/nodeserver.go]: we still need this

File: pkg/cephfs/nodeserver.go (hunk 2 above)
Comment (Madhu-1) [reply pkg/cephfs/nodeserver.go]: yes its required to fix gocyclo static chec

File: pkg/cephfs/nodeserver.go (hunk 3 above)
Comment (Madhu-1) [reply pkg/cephfs/nodeserver.go]: we don't need this one as the same function is implemented in CSI common https://github.com/kubernetes-csi/drivers/blob/master/pkg/csi-common/nodeserver-default.go#L39

File: pkg/rbd/controllerserver.go [hunk 4]
```diff
@@ -159,16 +168,38 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
...
-			VolumeId:      volumeID,
-			CapacityBytes: volSizeBytes,
+			VolumeId:      rbdVol.VolID,
+			CapacityBytes: rbdVol.VolSize,
 			VolumeContext: req.GetParameters(),
 		},
 	}, nil
 }
 
+func (cs *ControllerServer) checkrbdStatus(rbdVol *rbdVolume, req *csi.CreateVolumeRequest, volSizeGB int) error {
```
[the remark this answers is not in the document]
Comment (Madhu-1) [reply pkg/rbd/controllerserver.go]: done

**Comment by Madhu-1** [comment]
@rootfs PTAL

File: pkg/cephfs/nodeserver.go (hunk 3 above)
Comment (rootfs) [reply pkg/cephfs/nodeserver.go]: but we have to call the default implementation in our node server

Comment (Madhu-1) [reply pkg/cephfs/nodeserver.go]: @rootfs everything worked fine in during my testing let me check once again

Comment (Madhu-1) [reply pkg/cephfs/nodeserver.go]: @rootfs  default node server is getting called

```
I0129 13:59:10.161483       1 nodeserver-default.go:40] Using default NodeGetInfo
I0129 13:59:10.161504       1 utils.go:103] GRPC response: {"node_id":"kube2"}
```

**Comment by rootfs** [comment]
thanks, let's pick up the changes.

**Pull request merged**